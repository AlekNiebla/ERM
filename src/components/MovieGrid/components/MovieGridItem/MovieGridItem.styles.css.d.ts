export const main: string;
export const poster: string;
export const noPoster: string;
export const data: string;
export const title: string;
export const releaseDate: string;
export const genres: string;
