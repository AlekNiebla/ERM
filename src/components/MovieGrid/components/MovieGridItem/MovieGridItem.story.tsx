import * as React from "react";
import { storiesOf } from "@storybook/react";
import MovieGridItem, { IMovieGridItem } from "./MovieGridItem";
// mock data
const singleMovie: IMovieGridItem = {
  id: 1,
  title: "Pulp Fiction",
  poster: "https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
  releaseDate: "1994-09-10",
  genres: ["Thriller", "Crime"]
};

storiesOf("Components / MovieGrid / MovieGridItem", module).add(
  "default",
  () => (
    <div
      style={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        backgroundColor: "#f1f2f3",
        padding: "100px"
      }}
    >
      <MovieGridItem {...singleMovie} />
    </div>
  )
);
