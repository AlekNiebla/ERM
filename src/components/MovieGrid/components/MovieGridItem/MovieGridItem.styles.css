.main {
  border-radius: var(--step-3);
  overflow: hidden;
  max-width: 250px;
  border: 1px solid var(--color-gray-lighter);
  background-color: var(--color-white);
  box-shadow: 0 2px 5px 0 var(--color-gray-lighter);
  position: relative;
}
.poster {
  max-width: 100%;
  display: block;
  z-index: 1;
}
.noPoster {
  width: 100%;
  display: block;
  height: 100%;
  color: var(--color-gray-dark);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1;
}
.data {
  display: flex;
  flex-direction: column;
  padding: var(--step-3);
  position: absolute;
  bottom: 0;
  width: 100%;
  z-index: 4;
  background-color: var(--color-white-transparent);
}
.baseData {
  margin-bottom: var(--step-2);
}
.title {
  composes: Typo-Big-Bold from global;
  color: var(--color-gray-darkest);
  margin-right: var(--step-2);
}
.releaseDate {
  composes: Typo-Base from global;
  color: var(--color-primary);
}
.genres {
  composes: Typo-Small from global;
  color: var(--color-gray-dark);
}
