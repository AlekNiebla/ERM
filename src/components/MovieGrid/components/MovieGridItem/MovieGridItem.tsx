import * as React from "react";

const MovieGridItem = () => <span>MovieGridItem</span>;

export default MovieGridItem;
