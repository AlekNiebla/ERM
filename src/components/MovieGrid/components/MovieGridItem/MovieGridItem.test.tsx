import * as React from "react";
import { shallow } from "enzyme";
// components
import MovieGridItem from "./MovieGridItem";

describe("MovieGridItem component...", () => {
  it("...renders without errors", () => {
    const wrapper = shallow(<MovieGridItem />);

    expect(wrapper.find(MovieGridItem)).toBeTruthy();
  });
});
