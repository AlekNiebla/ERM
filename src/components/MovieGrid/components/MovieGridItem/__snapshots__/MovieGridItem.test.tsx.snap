// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGridItem component... ... matches snapshot 1`] = `ShallowWrapper {}`;

exports[`MovieGridItem component... ...renders without errors 1`] = `ShallowWrapper {}`;
